exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/CreateTableHandler.java

```diff
@@ -83,7 +84,20 @@ public PhysicalPlan getPlan(SqlNode sqlNode) throws ValidationException, RelConv
     final DrillConfig drillConfig = context.getConfig();
     final AbstractSchema drillSchema = resolveSchema(sqlCreateTable, config.getConverter().getDefaultSchema(), drillConfig);
 
-    checkDuplicatedObjectExistence(drillSchema, originalTableName, drillConfig, context.getSession());
+    String schemaPath = drillSchema.getFullSchemaName();
+
+    // Check duplicate object existence
+    boolean isTemporaryTable = context.getSession().isTemporaryTable(drillSchema, drillConfig, originalTableName);
```

Comment:
1. Please add unit tests for temp tables as well in `TestCTTAS` class.
2. Should we consider similar functionality for views?

Reply:
1) Added unit tests for temp tables
2) Implemented for views as well